lib/puppet/network/authconfig.rb

```diff
@@ -89,4 +89,28 @@ def initialize(rights=nil)
       insert_default_acl
     end
   end
+
+  class Network::AuthConfig
+    @@authprovider_class = nil
+
+    def self.authprovider_class=(klass)
+      @@authprovider_class = klass
+    end
+
+    def self.authprovider_class
+      @@authprovider_class || Puppet::Network::DefaultAuthProvider
+    end
+
+    def initialize(rights=nil)
+      if rights.nil?
```

Comment:
Do we need the `if` check here or can we just do `@authprovider = self.class.authprovider_class.new(rights)`?